File: src/main/java/ru/pgk/pgk/features/department/entitites/DepartmentEntity.java
package ru.pgk.pgk.features.department.entitites;

import jakarta.persistence.*;
import lombok.Data;
import ru.pgk.pgk.features.group.entities.GroupEntity;
import ru.pgk.pgk.features.schedule.entities.ScheduleEntity;
import ru.pgk.pgk.features.teacher.entities.TeacherEntity;

import java.io.Serializable;
import java.util.Collection;

@Data
@Entity(name = "departments")
public class DepartmentEntity implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Short id;

    private String name;

    @OneToMany(mappedBy = "department", fetch = FetchType.LAZY)
    private Collection<GroupEntity> groups;

    @ManyToMany(mappedBy = "departments", fetch = FetchType.LAZY)
    private Collection<TeacherEntity> teachers;

    @OneToMany(mappedBy = "department", fetch = FetchType.LAZY)
    private Collection<ScheduleEntity> schedules;
}

File: teacher-service/src/main/kotlin/ru/pgk/teacher_service/features/entities/DepartmentEntity.kt
package ru.pgk.teacher_service.features.entities

import jakarta.persistence.*
import java.io.Serializable

@Entity(name = "departments")
data class DepartmentEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int = 0,
    var name: String,

    @ManyToMany(mappedBy = "departments", fetch = FetchType.LAZY)
    val teachers: List<TeacherEntity> = emptyList()
): Serializable
